cpp/src/arrow/compute/function.h

```diff
@@ -159,6 +159,20 @@ struct ARROW_EXPORT FunctionDoc {
   static const FunctionDoc& Empty();
 };
 
+/// \brief An executor of a function with a preconfigured kernel
+struct ARROW_EXPORT FunctionExecutor {
+  virtual ~FunctionExecutor() = default;
+  /// \brief Initialize a preconfigured kernel
+  ///
+  /// This method may be called zero or more times. By default, the kernel is initialized
+  /// with default function options and exec context.
```

Comment:
```suggestion
  /// This method may be called zero or more times. Depending on how
  /// the FunctionExecutor was obtained, it may already have been initialized.
```